Lib/datetime.py

```diff
@@ -1631,8 +1701,20 @@ def _fromtimestamp(cls, t, utc, tz):
 
         A timezone info object may be passed in as well.
         """
-        frac, t = _math.modf(t)
-        us = round(frac * 1e6)
+        t, frac = divmod(_Decimal(t), _Decimal(1))
```

Comment:
None of this code involving `Decimal` is safe as written; an arbitrary `Decimal` context could be in effect, potentially reducing the precision of `Decimal` below that of even `float`.